### DISCUSSION TITLE
How to keep error state with `placeholderData` from `useQuery`?

### DISCUSSION THREAD

**Discussion by park-jemin** [TanStack/query#7629] [Q&A]
I'm using `placeholderData` to maintain the previous data when transitioning between pages, which defaults it to a success state. However, when I hit an error from a fetch call, I would like to _keep_ the error state instead of keeping the previous data when I transition, until the next time I hit a successful fetch or cache hit. 

The flow I'm running into rn looks kinda like this:
- Call `useQuery` with `page = 1`, returns a successful response
- Call `useQuery` with `page = 2`, returns a successful response while temporarily using the previous page data as a placeholder
- Call `useQuery` with `page = 3`, returns an error response (with `isError = true`) while temporarily using the previous page data as a placeholder
- Call `useQuery` with `page = 4`, returns an error response, but temporarily flashes the page 2 data as a placeholder (with `isError` temporarily being set to `true`)

How should I go about approaching this?

**Comment by TkDodo** [comment]
I don't think its possible to keep the error. You can implement `placeholderData` to return undefined if the query is in error state to avoid the flash.
